### PR TITLE
Added networking

### PR DESCRIPTION
**Pull request by zakrent**

### CODE REVIEW

File: src/structs.rs [hunk 1]
```diff
@@ -1,18 +1,22 @@
 use bit_vec::BitVec;
 use std::collections::HashMap;
 
+#[derive(Serialize, Deserialize, Debug)]
 pub struct Map {
     width: usize,
     height: usize,
-    terrain: BitVec,
+    terrain: Vec<bool>,
```
Comment (Bravo555) [inline src/structs.rs:8]: Is this intentional, or you have an old version of the code?

**Review by Bravo555 — CHANGES_REQUESTED** [review]
Looks ok, install [rustfmt](https://github.com/rust-lang/rustfmt) pls. `Bitvec` is why I'm not passing it rn tho.

File: src/structs.rs (hunk 1 above)
Comment (zakrent) [reply src/structs.rs:8]: Serde doesn't support BitVec yet, we have to implement it ourselves (which I don't know how to do) or use Vec.

**Comment by zakrent** [comment]
Now it's a final version, won't push anything except potential requested changes.

File: src/structs.rs (hunk 1 above)
Comment (Bravo555) [reply src/structs.rs:8]: Okay, makes sense. Good to go then.

**Review by Bravo555 — APPROVED** [review]

**Pull request merged**